internal/providers/terraform/aws/db_instance.go

```diff
@@ -192,6 +192,34 @@ func NewDBInstance(d *schema.ResourceData, u *schema.UsageData) *schema.Resource
 		})
 	}
 
+	var backupStorageGB *decimal.Decimal
```

Comment:
Should I wrap this in a `if d.Get("backup_retention_period).Exists() { }` to check that `backup_retention_period` is configured on the `aws_db_instance` resource?

Reply:
Good idea! Maybe we can use the following condition so we also show the cost component if a user has explicitly set the usage param (e.g. to do what-if analysis before even enabling the Terraform param).

```
if d.Get("backup_retention_period).Exists() || (u != nil && u.Get("additional_backup_storage_gb").Exists())
```

---

This isn't required in this PR, but we have an [`IsEmpty` func](https://github.com/infracost/infracost/pull/1135#issuecomment-970499598) that we can extract and re-use instead of `Exists()` (as it catches some rare conditions we've seen in TF plan JSONs). @vdmgolub any idea where we should put this function so it can be re-used for `d.` and `u.` checks?

Reply:
@alikhajeh1 @joshniec Both `ResourceData` (d, [link](https://github.com/infracost/infracost/blob/master/internal/schema/resource_data.go#L49-L54)) and `UsageData` (u, [link](https://github.com/infracost/infracost/blob/master/internal/schema/usage_data.go#L74-L79)) structs have this `IsEmpty(key)` function (identical code in both places). I guess that'll do for now.

Reply:
Thanks, I went ahead and refactored my code to make use of the `IsEmpty(key)` function. I also replaced other calls of `d.Get(key).Exists()` and `u.Get(key).Exists()` to the `IsEmpty(key)` function in the same file (which let me remove the `"github.com/tidwall/gjson"` import).